resident/resident-service/src/main/java/io/mosip/resident/exception/ApiExceptionHandler.java

```diff
@@ -120,6 +120,13 @@ private ResponseEntity<ResponseWrapper<ServiceError>> createResponseEntity(
 			headers.add(ResidentConstants.EVENT_ID,
 					(String) ((ObjectWithMetadata) e).getMetadata().get(ResidentConstants.EVENT_ID));
 			return new ResponseEntity<>(errorResponse, headers, httpStatus);
+		} else if (e instanceof ObjectWithMetadata && ((ObjectWithMetadata) e).getMetadata() != null

```

Comment:
Please check each conditions separately and set them individually, it should not fail entirely if one condition fails.

Reply:
changes done